File: Level 2/괄호 회전하기/solution.js
function solution(s) {
  let answer = 0;
  const mapping = { "}": "{", "]": "[", ")": "(" };

  for (let i = 0; i < s.length; i++) {
    const stack = [];
    const rotatedS = s.slice(i) + s.slice(0, i);

    for (let j = 0; j < s.length; j++) {
      if (rotatedS[j] === "[" || rotatedS[j] === "(" || rotatedS[j] === "{") {
        stack.push(rotatedS[j]);
        continue;
      }

      if (stack[stack.length - 1] === mapping[rotatedS[j]]) {
        stack.pop();
      } else {
        stack.push(rotatedS[j]);
      }
    }

    if (stack.length === 0) {
      answer++;
    }
  }

  return answer;
}

File: Level 2/괄호 회전하기/solution.py
closeBraketMap = {"}": "{", "]": "[", ")": "("}


def solution(s):
    answer = 0

    for i in range(len(s)):
        rotatedS = s[i:] + s[:i]

        stack = []

        for braket in rotatedS:
            if braket in ("{", "[", "("):
                stack.append(braket)
            else:
                if stack and stack[-1] == closeBraketMap[braket]:
                    stack.pop()
                else:
                    break
        else:
            if not stack:
                answer += 1

    return answer
